src/JSInterop/Microsoft.JSInterop/src/Infrastructure/DotNetObjectReferenceJsonConverterFactory.cs

```diff
@@ -22,6 +22,7 @@ public override bool CanConvert(Type typeToConvert)
     }
 
     [UnconditionalSuppressMessage("Trimming", "IL2055", Justification = "We expect that types used with DotNetObjectReference are retained.")]
+    [SuppressMessage("AOT", "IL3050:Calling members annotated with 'RequiresDynamicCodeAttribute' may break functionality when AOT compiling.", Justification = "MakeGenericType is AOT safe for reference types.")]
```

Comment:
Is this always used with reference types?

Add a `Debug.Assert(instanceType.IsClass)` to verify.